public/app/features/alerting/unified/components/rules/AlertInstancesTable.tsx

```diff
@@ -20,6 +22,18 @@ export const AlertInstancesTable: FC<Props> = ({ instances }) => {
 
   const [expandedKeys, setExpandedKeys] = useState<string[]>([]);
 
+  // add key & sort instance. API returns instances in random order, different every time.
```

Comment:
I just noticed this as well! Instances would jump around every time the polling interval was hit.